add-on/panels/js/panel.jsx

```diff
@@ -984,6 +976,119 @@ loop.panel = (function(_, mozL10n) {
     }
   });
 
+  var SharePanelView = React.createClass({
+    statics: {
+      SHOW_PANEL_DELAY: 300
+    },
+
+    mixins: [
+      Backbone.Events,
+      sharedMixins.DocumentVisibilityMixin,
+      sharedMixins.WindowCloseMixin
+    ],
+
+    propTypes: {
+      dispatcher: React.PropTypes.instanceOf(loop.Dispatcher).isRequired,
+      // Force render for testing purpose
+      forceRender: React.PropTypes.bool,
+      store: React.PropTypes.instanceOf(loop.store.RoomStore).isRequired
+    },
+
+    getInitialState: function() {
+      return this.props.store.getStoreState();
+    },
+
+    componentWillMount: function() {
+      loop.request("GetLoopPref", "facebook.enabled").then(result => {
+        this.setState({
+          facebookEnabled: result
+        });
+      });
+    },
+
+    componentDidMount: function() {
+      this.listenTo(this.props.store, "change", this._onStoreStateChanged);
+    },
+
+    componentDidUpdate: function() {
+      if (this.state.showPanel) {
+        setTimeout(() => {
+          this.getDOMNode().classList.add("share-panel-open");
+        }, this.constructor.SHOW_PANEL_DELAY);
+      }
+    },
+
+    componentWillUnmount: function() {
+      this.stopListening(this.props.store);
+    },
+
+    _onStoreStateChanged: function() {
+      this.setState(this.props.store.getStoreState());
+    },
+
+    onDocumentHidden: function() {
+      this.state.showPanel && this.handleClosePanel();
+    },
+
+    componentWillUpdate: function(nextProps, nextState) {
+      // If we've just created a room, close the panel - the store will open
+      // the room.
```

Comment:
The comment here doesn't seem to make sense wrt the code - the code is getting the panel to open, whereas this is talking about closing the panel.